extensions/scarb-doc/src/db.rs

```diff
@@ -0,0 +1,120 @@
+use cairo_lang_compiler::project::{
+    update_crate_root, update_crate_roots_from_project_config, ProjectConfig,
+};
+use cairo_lang_defs::db::{DefsDatabase, DefsGroup};
+use cairo_lang_doc::db::{DocDatabase, DocGroup};
+use cairo_lang_filesystem::db::{
+    init_files_group, AsFilesGroupMut, FilesDatabase, FilesGroup, CORELIB_CRATE_NAME,
+};
+use cairo_lang_parser::db::{ParserDatabase, ParserGroup};
+use cairo_lang_semantic::db::{SemanticDatabase, SemanticGroup};
+use cairo_lang_semantic::inline_macros::get_default_plugin_suite;
+use cairo_lang_semantic::plugin::PluginSuite;
+use cairo_lang_starknet::starknet_plugin_suite;
+use cairo_lang_syntax::node::db::{SyntaxDatabase, SyntaxGroup};
+use cairo_lang_utils::Upcast;
+
+use salsa;
+
+/// The Cairo compiler Salsa database tailored for scarb-doc usage.
+#[salsa::database(
+    FilesDatabase,
+    ParserDatabase,
+    SyntaxDatabase,
+    DefsDatabase,
+    SemanticDatabase,
+    DocDatabase
+)]
+pub struct ScarbDocDatabase {
+    storage: salsa::Storage<Self>,
+}
+
+impl ScarbDocDatabase {
+    pub fn new(project_config: Option<ProjectConfig>) -> Self {
+        let mut db = Self {
+            storage: Default::default(),
+        };
+
+        init_files_group(&mut db);
+
+        let plugin_suite = [get_default_plugin_suite(), starknet_plugin_suite()]
```

Comment:
consider leaving a note somewhere that scarb-provided plugins have to be supported. I'd just put `// FIXME(piotmag769): blahblah` here

Reply:
I think this should be a whole issue not TODO 😅

Reply:
True dat

Reply:
https://github.com/software-mansion/scarb/issues/1412